pkg/controllers/state/cluster.go

```diff
@@ -307,12 +313,40 @@ func (c *Cluster) UpdatePod(ctx context.Context, pod *corev1.Pod) error {
 	return err
 }
 
+// SetPodAckTime marks the pod as validating for scheduling. This is used to construct metrics
+// when the pod actually schedules to when Karpenter actually thought it could schedule.
+// This allows users to understand how long it took for Karpenter to actually start scheduling
+// for this pod.
+func (c *Cluster) SetPodAckTime(pods ...*corev1.Pod) {
+	for _, pod := range pods {
+		// Check the pod to see if we've already emitted a metric for it, and just continue
+		nn := types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}
+		c.mu.RLock()
```

Comment:
This mutex is used for state nodes -- do we want to take another mutex for this data since this data is unrelated to the node data -- and I don't think we want the speed of grabbing state nodes to be affected by setting this observability metric